src/test/java/bio/terra/common/fixtures/ConnectedOperations.java

```diff
@@ -173,7 +173,7 @@ public DatasetSummaryModel createDataset(DatasetRequestModel datasetRequest) thr
             .andReturn();
         MockHttpServletResponse response = validateJobModelAndWait(result);
         DatasetSummaryModel datasetSummaryModel = handleSuccessCase(response, DatasetSummaryModel.class);
-        addDataset(datasetSummaryModel.getId());
+        addDataset(datasetSummaryModel.getId().toString());
```

Comment:
You've already done so much, so I kind of don't want to add more, but I think if you change the underlying storage model of `ConnectedOperations` from `String` to `UUID`, it would be easier to get this working, and result in less code overall. This would mean migrating `ConnectedOperations.createdSnapshotIds`, `createdDatasetIds`, and all other fields that are tracking UUIDs as strings.

Reply:
Sorry I got side-tracked from this PR! I made those updates and I agree it makes the code clearer. Thanks for the suggestion!